src/Service/EmailService.php

```diff
@@ -0,0 +1,94 @@
+<?php
+namespace App\Service;
+
+use App\Entity\User;
+use Aws\Ses\SesClient;
+use Aws\Exception\AwsException;
+use Doctrine\Common\Persistence\ObjectManager;
+
+class EmailService {
+
+    /**
+     * @var \Swift_Mailer
+     *
+     */
+    private $mailer;
+    private $templating;
+    private $manager;
+
+    public function __construct(\Swift_Mailer $mailer, \Twig_Environment $templating,ObjectManager $manager)
+    {
+        $this->mailer = $mailer;
+        $this->templating = $templating;
+        $this->manager = $manager;
+    }
+
+    //Email function for sending the confirmation email
+    public function confirmRegistration($username, $email, $token)
+    {
+        $message = (new \Swift_Message('Confirmez votre adresse email'))
+            ->setFrom(getenv('MAILER_FROM'))
+            ->setTo($email)
+            ->setBody(
+                $this->templating->render(
+                    'emails/registration.html.twig',
+                    ['username' => $username,
+                        'token' => $token]
+                ),
+                'text/html'
+            );
+        $this->mailer->send($message);
```

Comment:
`$this->mailer->send()` should return something (either a success or fail boolean, or maybe the number of mails sent, or something else I'm not sure, but it has to be returning some kind of information) that we can use to say whether or not the mail was successfully sent. If it returns a boolean then just return it directly, if it returns the number of sent mails, check that it's not 0 etc. In short: return a boolean so we know if the mail was sent or not.

Reply:
Actually I don't find in the swiftMailer documentation something about email sending response ...

Reply:
okay, the service now return result for sending Email, it returns error if the email does not exist. The signup is cancelled if this function return false. Need to dev ASAP exception and front explanation for the user